### PR TITLE
Moving Docs to a GitHub Page

### PR DESCRIPTION
**Pull request by hayleycd**
Based on conversation in issue #201 I have set up a GitHub page for OSV-Scanner's documentation. 

Notes:

- You can see the rendered documentation [here](hayleycd.github.io/osv-scanner). 
- Docs are divided into current and working. All docs are present on the website, but [working docs](https://hayleycd.github.io/osv-scanner/working/) do not have their urls in the nav. 
- This is largely a structural change to the docs, not a content change to the docs. Changes were made to READMEs and other files, but generally this change is structural.

### CODE REVIEW

File: docs/_posts/2023-02-04-welcome-to-jekyll.markdown [hunk 1]
```diff
@@ -0,0 +1,29 @@
+---
```
Comment (another-rex) [inline docs/_posts/2023-02-04-welcome-to-jekyll.markdown:1]: Should this file/folder be removed?

File: docs/current/contribute.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+---
+layout: page
+title: Contribute
+permalink: /contribute/
+nav_order: 6
+---
+## Contribute
+
+### Report Problems
+If you have what looks like a bug, please use the [Github issue tracking system](https://github.com/ossf/scorecard/issues). Before you file an issue, please search existing issues to see if your issue is already covered.
```
Comment (another-rex) [inline docs/current/contribute.md:10]: This github tracking link was wrong when 1.1 released, use the working docs link here instead.
```suggestion
If you have what looks like a bug, please use the [Github issue tracking system](https://github.com/google/osv-scanner/issues). Before you file an issue, please search existing issues to see if your issue is already covered.
```

File: docs/current/contribute.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+### Contributing code to `osv-scanner`
+
+See [CONTRIBUTING.md](CONTRIBUTING.md) for documentation on how to contribute code.
```
Comment (another-rex) [inline docs/current/contribute.md:14]: This probably needs to be a full link to the Github file.

File: docs/current/installation.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+```console
+go install github.com/google/osv-scanner/cmd/osv-scanner@v1
+```
+
+This requires Go 1.18+ to be installed.
+
+### Build from source
+
+See [CONTRIBUTING.md](CONTRIBUTING.md) file.
```
Comment (another-rex) [inline docs/current/installation.md:45]: Probably needs to be full link to the github repo file

File: docs/working_docs/usage.md [hunk 5]
```diff
@@ -0,0 +1,126 @@
+---
+layout: default
+title: Usage
+nav_order: 2
+nav_exclude: true
+parent: Working
+permalink: /working/usage
+---
+{: .warning }
+These docs are a work in progress! You are welcome to browse, but you may find features/options that are not compatible with the currently released version of OSV-Scanner. Current OSV-Scanner usage docs can be found [here](../usage).
```
Comment (another-rex) [inline docs/working_docs/usage.md:10]: This part seems to be repeated a lot, it seems like Jekyll has a includes function, can you have a look and see if that can be used here? https://jekyllrb.com/docs/includes/

File: docs/README.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+[pre-requisites]: https://docs.github.com/en/pages/setting-up-a-github-pages-site-with-jekyll/testing-your-github-pages-site-locally-with-jekyll#prerequisites
+[docs locally]: https://docs.github.com/en/pages/setting-up-a-github-pages-site-with-jekyll/testing-your-github-pages-site-locally-with-jekyll#building-your-site-locally
+
+## Contributing to the docs
+Documentation contributions are welcome!
+
+Contributors should make changes to the documentation in [working docs](working_docs/).
+
+See [CONTRIBUTING.md](../CONTRIBUTING.md) for contribution guidelines.
```
Comment (another-rex) [inline docs/README.md:19]: Can you update CONTRIBUTING.md with a line or two about contributing to docs?

Also full link to github repo file is probably required here.

**Review by another-rex — COMMENTED** [review]
Thanks! This is looking very good, just added some initial comments.

File: docs/working_docs/output.md [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+╰─────────────────────────────────────┴───────────┴──────────────────────────┴─────────┴────────────────────╯
+```
+
+### `json` format
+
+Outputs the results as a JSON object to stdout, with all other output being directed to stderr - this makes it safe to redirect the output to a file with `osv-scanner --format json ... > /path/to/file.json`.
+
+Sample output:
+
+```json5
```
Comment (another-rex) [inline docs/working_docs/output.md:37]:
```suggestion
```json
```
Seems like the default code highlighter doesn't support json5 as an option, use json instead. (https://github.com/rouge-ruby/rouge/wiki/List-of-supported-languages-and-lexers)

File: docs/current/output.md [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+nav_order: 5
+---
+## JSON output
+By default osv-scanner outputs a human readable table. To have osv-scanner output JSON instead, pass the `--json` flag when calling osv-scanner.
+
+When using the --json flag, only the JSON output will be printed to stdout, with all other outputs being directed to stderr. So to save only the json output to file, you can redirect the output with `osv-scanner --json ... > /path/to/file.json`
+
+### Output Format
+
+```json5
```
Comment (another-rex) [inline docs/current/output.md:14]:
```suggestion
```json
```

File: docs/working_docs/usage.md (hunk 5 above)
Comment (hayleycd) [reply docs/working_docs/usage.md:10]: Thanks, I'll investigate.

File: docs/_posts/2023-02-04-welcome-to-jekyll.markdown (hunk 1 above)
Comment (hayleycd) [reply docs/_posts/2023-02-04-welcome-to-jekyll.markdown:1]: Agreed.

File: docs/README.md (hunk 6 above)
Comment (hayleycd) [reply docs/README.md:21]: Done! This link works because it is in the readme, while the others did not because they were on the github page.

**Review by another-rex — APPROVED** [review]
LGTM!

**Review by oliverchang — APPROVED** [review]

**Pull request merged**